extensions/wikia/AdEngine/js/template/roadblock.js

```diff
@@ -0,0 +1,89 @@
+/*global define, require*/
+define('ext.wikia.adEngine.template.roadblock', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.context.uapContext',
+	'ext.wikia.adEngine.provider.gpt.googleSlots',
+	'ext.wikia.adEngine.provider.gpt.helper',
+	require.optional('ext.wikia.adEngine.template.skin'),
+	'wikia.document',
+	'wikia.window',
+	'wikia.log'
+], function (
+	adContext,
+	uapContext,
+	googleSlots,
+	gptHelper,
+	skinTemplate,
+	doc,
+	win,
+	log
+) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.template.roadblock',
+		uapType = 'ruap',
+		medrecSlotEl = doc.getElementById('TOP_RIGHT_BOXAD');
+
+	/**
+	 * Handles medrec slot in case of UAP:roadblock
+	 * @param {Node} medrecSlotEl - medrec slot container element
+	 */
+	function handleMedrec(medrecSlotEl, uapId) {
+		var refreshed = false;
+
+		win.addEventListener('adengine.slot.status', function onAdStatusChange(event) {
+			var gptParams = {},
+				isMedrec = event.detail.slot.name === medrecSlotEl.id,
+				isSuccess = event.detail.adInfo.status === 'success',
```

Comment:
I'm not sure if success is the only event you need to bind on.  What about `hop`, `collapse`? Maybe even `renderEnded`?